crates/rolldown/src/stages/link_stage/mod.rs

```diff
@@ -150,16 +150,11 @@ impl<'a> LinkStage<'a> {
             if matches!(importee.exports_kind, ExportsKind::None)
               && !importee.meta.has_lazy_export()
             {
-              // See https://github.com/evanw/esbuild/issues/447
-              if rec.meta.intersects(
-                ImportRecordMeta::CONTAINS_IMPORT_DEFAULT | ImportRecordMeta::CONTAINS_IMPORT_STAR,
```

Comment:
There are useless now. Wonder if we should remove them too. Took me quite of time to add them :(.